zeppelin-web/src/components/noteName-create/visible.directive.js

```diff
@@ -0,0 +1,35 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+angular.module('zeppelinWebApp').directive('modalvisible', function () {
+    return {
+        restrict: 'A',
+        scope: {
+        	preVisibleCallback: '&previsiblecallback',
+        	postVisibleCallback: '&postvisiblecallback'	},
+        link: function(scope, elem, attrs) {
+        	// Add some listeners
+    		var previsibleMethod = scope.preVisibleCallback;
+    		var postVisibleMethod = scope.postVisibleCallback;
+    		elem.on('show.bs.modal',function(e) {
+    			previsibleMethod();
+    		});
+    		elem.on('shown.bs.modal', function(e) {
+    			$(e.target).find('#noteName').select();
```

Comment:
Only change left I can see, is passing the id of the element to the directive, and do the select if the id was passed.